src/client/App.jsx

```diff
@@ -1,60 +1,92 @@
 import { Competition } from './components/Competition';
 import { Season } from './components/Season';
 
+import { API_URL, HTTP_METHOD, LOCAL_STORAGE } from './config';
+
 import './App.css';
 
 const App = () => {
-    const [isLoggedIn, setIsLoggedIn] = useState(true);
-    const [user, setUser] = useState("nathan");
+    const [isLoggedIn, setIsLoggedIn] = useState(false);
+    const [user, setUser] = useState(null);
+
+    const navigate = useNavigate();
 
-    const findUser = () => {
-        setIsLoggedIn(true);
-    };
+    useEffect(() => {
+        localStorage.getItem(LOCAL_STORAGE.TOKEN) ? setIsLoggedIn(true) : setIsLoggedIn(false);
+    }, []);
 
     useEffect(() => {
-        findUser();
-    });
+        if (!isLoggedIn) return;
+
+        const authenticateUser = async () => {
+            try {
+                const response = await fetch(API_URL.GET, {
+                    method: HTTP_METHOD.GET,
+                    headers: {
+                        Authorization: localStorage.getItem(LOCAL_STORAGE.TOKEN),
+                    },
+                });
+
+                const result = await response.json();
+
+                if (result) {
+                    setUser(result.data.username);
+                }
+            } catch (error) {
+                console.log(error);
+            }
+        };
+
+        authenticateUser();
+    }, [isLoggedIn]);
 
     useEffect(() => {
-        //get token from local storage
-        //decode token, get id
-        //fetch user then setUser
-        
-    },[isLoggedIn]);
+        navigate(`/${user}`);
+    }, [user]);
 
     return (
         <div className="app">
             <Header
                 isLoggedIn={isLoggedIn}
                 user={user}
                 setUser={setUser}
+                setIsLoggedIn={setIsLoggedIn}
             />
-
-
             <Routes>
-                {!isLoggedIn &&
-                    <Route path="/"
-                        element={<Welcome setIsLoggedIn={setIsLoggedIn}/>}
-                    />
-                }
-                {isLoggedIn && <>
-                    <Route path="/:user"
-                        element={<Homepage user={user}/>}
-                    />
-                    <Route path="/:user/:competitionId"
-                        element={<Competition user={user}/>}
-                    />
-                    <Route path="/:user/:competitionId/:seasonId"
-                        element={<Season user={user}/>}
-                    />
-                </>}
+                {!isLoggedIn && (
+                    <>
+                        <Route
+                            path="/"
+                            element={<Welcome setIsLoggedIn={setIsLoggedIn} />}
+                        />
+                        <Route
+                            path="*"
+                            element={<Welcome setIsLoggedIn={setIsLoggedIn} />}
+                        />
+                    </>
+                )}
```

Comment:
What are the separate components for?

Reply:
if a user goes to a link that isn't defined it get routed through '*' which renders the homepage. Its a quick fix in the absence of a 404 page. That's a task for a different issue/branch.